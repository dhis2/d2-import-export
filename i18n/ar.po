# Translators:
# Viktor Varland <viktor@dhis2.org>, 2019
# phil_dhis2, 2019
# Hamza Assada <7amza.it@gmail.com>, 2020
# 
msgid ""
msgstr ""
"Project-Id-Version: i18next-conv\n"
"POT-Creation-Date: 2020-01-07T10:27:09.944Z\n"
"PO-Revision-Date: 2019-09-02 07:17+0000\n"
"Last-Translator: Hamza Assada <7amza.it@gmail.com>, 2020\n"
"Language-Team: Arabic (https://www.transifex.com/hisp-uio/teams/100509/ar/)\n"
"MIME-Version: 1.0\n"
"Content-Type: text/plain; charset=UTF-8\n"
"Content-Transfer-Encoding: 8bit\n"
"Language: ar\n"
"Plural-Forms: nplurals=6; plural=n==0 ? 0 : n==1 ? 1 : n==2 ? 2 : n%100>=3 && n%100<=10 ? 3 : n%100>=11 && n%100<=99 ? 4 : 5;\n"

msgid "user is not logged in"
msgstr "لم يتم تسجيل دخول المستخدم"

msgid "Import Error"
msgstr "تصدير الخطأ"

msgid "Go Back"
msgstr "العودة للخلف"

msgid "filter data sets by name"
msgstr "تصفية حزم البيانات حسب الإسم"

msgid "Select All"
msgstr "تحديد الكل"

msgid "Clear All"
msgstr "الغاء التحديد عن الكل"

msgid "Data sets"
msgstr "حزم البيانات"

msgid "Choose a file to upload"
msgstr "اختر ملفاً للتحميل"

msgid "more options"
msgstr "المزيد من الخيارات"

msgid "JSON"
msgstr ""

msgid "XML"
msgstr ""

msgid "CSV"
msgstr ""

msgid "Select None"
msgstr "الغاء التحديد عن الكل"

msgid "Logger"
msgstr ""

msgid "view messages received on using import/export forms."
msgstr ""

msgid "Import"
msgstr "استيراد"

msgid "Export"
msgstr "تصدير"

msgid "Created"
msgstr "تم الإنشاء"

msgid "Deleted"
msgstr "تم حذفه"

msgid "Ignored"
msgstr "تم تجاهله"

msgid "Updated"
msgstr "تم تحديثه"

msgid "Total"
msgstr "الإجمالي"

msgid "Type"
msgstr "النوع"

msgid "Type Count"
msgstr ""

msgid "Messages"
msgstr "الرسائل"

msgid "Summaries"
msgstr "الخلاصات"

msgid "Conflicts"
msgstr "تعارضات"

msgid "Errors"
msgstr "اخطاء"

msgid "Import Summary"
msgstr "خلاصة عملية الإستيراد"

msgid "Summary"
msgstr "الخلاصة"

msgid "Async"
msgstr "غير متزامن"

msgid "Atomic Mode"
msgstr ""

msgid "Category ID scheme"
msgstr ""

msgid "Children"
msgstr ""

msgid "Object type"
msgstr "نوع العنصر"

msgid "Compression"
msgstr "الضغط"

msgid "Data element ID scheme"
msgstr ""

msgid "Dry run"
msgstr "تجريب دون حفظ فعلي للبيانات المستوردة"

msgid "End date"
msgstr "تاريخ النهاية"

msgid "Event ID Scheme"
msgstr ""

msgid "Flush Mode"
msgstr ""

msgid "Format"
msgstr "التنسيق"

msgid "ID scheme"
msgstr ""

msgid "Identifier"
msgstr "المعرف"

msgid "Report Mode"
msgstr ""

msgid "Import Strategy"
msgstr "استراتيجية الاستيراد"

msgid "Include deleted"
msgstr "تضمين ماتم حذفه"

msgid "Inclusion"
msgstr ""

msgid "Inclusion Strategy"
msgstr "استراتيجية الإدماج"

msgid "Merge Mode"
msgstr ""

msgid "Object"
msgstr "الكائن"

msgid "Organisation unit"
msgstr "الوحدة التنظيمية"

msgid "Org unit ID scheme"
msgstr ""

msgid "Preheat cache"
msgstr ""

msgid "Preheat Mode"
msgstr ""

msgid "Program Stages"
msgstr "مراحل البرنامج"

msgid "Programs"
msgstr "البرامج"

msgid "Schemas"
msgstr ""

msgid "Data Sets"
msgstr "ضبط البيانات"

msgid "Sharing"
msgstr "المشاركة"

msgid "Existing record check"
msgstr ""

msgid "Skip Sharing"
msgstr "تخطي بيانات المشاركة"

msgid "Skip Validation"
msgstr ""

msgid "Start date"
msgstr "تاريخ البداية"

msgid "Strategy"
msgstr "استراتيجية"

msgid "Yes"
msgstr "نعم"

msgid "No"
msgstr "لا"

msgid "UID"
msgstr "UID"

msgid "Code"
msgstr "الرمز"

msgid "Name"
msgstr "الاسم"

msgid "Include descendants of organisation unit"
msgstr ""

msgid "Selected organisation unit"
msgstr "الوحدة التنظيمية المختارة"

msgid "Yes (faster for large imports)"
msgstr "نعم (أسرع لعمليات الإستيراد الكبيرة)"

msgid "New and Updates"
msgstr "جديد وتحديثات"

msgid "New only"
msgstr "الجديد فقظ"

msgid "Updates only"
msgstr "تحديثات فقط"

msgid "Delete"
msgstr "حذف"

msgid "Skip check (fast)"
msgstr ""

msgid "Check (safe, recommended)"
msgstr "تحقق (آمن ، موصى به)"

msgid "AUTO"
msgstr ""

msgid "Full"
msgstr "كامل"

msgid "Debug"
msgstr ""

msgid "Reference"
msgstr "المرجع"

msgid "All"
msgstr "الجميع"

msgid "None"
msgstr "لا شيء"

msgid "Create and Update"
msgstr "إنشاء وتحديث"

msgid "Create"
msgstr "إنشاء"

msgid "Update"
msgstr "تحديث"

msgid "Merge"
msgstr "دمج"

msgid "Replace"
msgstr "استبدال"

msgid "Auto"
msgstr "تلقائي"

msgid "With sharing"
msgstr "مع بيانات المشاركة"

msgid "Without sharing"
msgstr "من غير بيانات المشاركة"

msgid "Non Null"
msgstr ""

msgid "Always"
msgstr "دائماً"

msgid "Non Empty"
msgstr "نعم (اسرع عند عمليات التصدير كبيرة الحجم)"

msgid "Include children of organisation unit"
msgstr ""

msgid "Zip"
msgstr ""

msgid "Gzip"
msgstr ""

msgid "Uncompressed"
msgstr "غير مضغوط"

msgid "Category combination"
msgstr "مزيج الفئات"

msgid "Dashboard"
msgstr "لوحة المعلومات"

msgid "Data element groups"
msgstr "مجموعات عناصر البيانات"

msgid "[ All program stages]"
msgstr "[كل مراحل البرنامج]"

msgid "Data Export"
msgstr "تصدير البيانات"

msgid ""
"Export data values. This is the regular export function which exports data "
"to the DHIS 2 exchange format called DXF 2."
msgstr ""

msgid "Loading options..."
msgstr "تحميل الخيارات..."

msgid "Event Export"
msgstr "تصدير الأحداث"

msgid ""
"Export event data for programs, stages and tracked entities in the DXF 2 "
"format."
msgstr ""
"تصدير الأحداث للبرامج والمراحل والكيانات المتعقبة باستخدام تنسيق DXF 2."

msgid "Metadata Export"
msgstr "تصدير البيانات الوصفية"

msgid ""
"Export meta data like data elements and organisation units to the standard "
"DHIS 2 exchange format."
msgstr ""

msgid "Meta Data Export"
msgstr ""

msgid "Metadata Dependency Export"
msgstr ""

msgid ""
"Export metadata like data sets and programs including related metadata "
"objects."
msgstr ""

msgid "Metadata Export with Dependencies"
msgstr ""

msgid "Import / Export"
msgstr "استيراد / تصدير"

msgid "Import or Export DHIS2 data."
msgstr ""

msgid "Data Import"
msgstr "استيراد البيانات"

msgid ""
"Import data values on the DXF 2 XML, JSON, CSV and PDF formatrant s. DXF 2 "
"is the standard exchange format for DHIS 2."
msgstr ""

msgid "Event Import"
msgstr "استيراد الأحداث"

msgid ""
"Import events for programs, stages and tracked entities in the DXF 2 format."
msgstr ""
"استيراد الأحداث للبرامج والمراحل والكيانات المتعقبة باستخدام تنسيق DXF 2."

msgid "GML Import"
msgstr "استيراد GML"

msgid ""
"Import geographic data for organisation units using GML format. GML is an "
"XML grammar for expressing geographical features."
msgstr ""
"استيراد البيانات الجغرافية للوحدات التنظيمية باستخدام تنسيق GML. GML عبارة "
"عن قواعد XML للتعبير عن المعالم الجغرافية."

msgid "Only import of GML data for existing organisation units is supported."
msgstr ""

msgid "Metadata Import"
msgstr "استيراد البيانات الوصفية"

msgid ""
"Import metadata like data elements and organisation units using the standard"
" DHIS 2 exchange format called DXF 2."
msgstr ""

msgid "It seems like there was no data in the uploaded file."
msgstr "يبدو أنه لا توجد بيانات في الملف الذي تم تحميله."
