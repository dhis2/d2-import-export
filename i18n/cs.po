# 
# Translators:
# trendspotter, 2020
# 
msgid ""
msgstr ""
"Project-Id-Version: i18next-conv\n"
"POT-Creation-Date: 2019-12-19T10:07:59.315Z\n"
"PO-Revision-Date: 2019-09-02 07:17+0000\n"
"Last-Translator: trendspotter, 2020\n"
"Language-Team: Czech (https://www.transifex.com/hisp-uio/teams/100509/cs/)\n"
"MIME-Version: 1.0\n"
"Content-Type: text/plain; charset=UTF-8\n"
"Content-Transfer-Encoding: 8bit\n"
"Language: cs\n"
"Plural-Forms: nplurals=4; plural=(n == 1 && n % 1 == 0) ? 0 : (n >= 2 && n <= 4 && n % 1 == 0) ? 1: (n % 1 != 0 ) ? 2 : 3;\n"

msgid "user is not logged in"
msgstr ""

msgid "Import Error"
msgstr ""

msgid "Go Back"
msgstr ""

msgid "filter data sets by name"
msgstr ""

msgid "Select All"
msgstr ""

msgid "Clear All"
msgstr ""

msgid "Data sets"
msgstr "Datové sady"

msgid "Choose a file to upload"
msgstr ""

msgid "Required"
msgstr ""

msgid "more options"
msgstr ""

msgid "JSON"
msgstr ""

msgid "XML"
msgstr ""

msgid "CSV"
msgstr ""

msgid "Select None"
msgstr ""

msgid "Logger"
msgstr ""

msgid "view messages received on using import/export forms."
msgstr ""

msgid "Import"
msgstr ""

msgid "Export"
msgstr ""

msgid "Created"
msgstr "Vytvořeno"

msgid "Deleted"
msgstr ""

msgid "Ignored"
msgstr ""

msgid "Updated"
msgstr ""

msgid "Total"
msgstr ""

msgid "Type"
msgstr "Typ"

msgid "Type Count"
msgstr ""

msgid "Messages"
msgstr "Zprávy"

msgid "Summaries"
msgstr ""

msgid "Conflicts"
msgstr ""

msgid "Errors"
msgstr "Chyby"

msgid "Import Summary"
msgstr ""

msgid "Summary"
msgstr ""

msgid "Async"
msgstr ""

msgid "Atomic Mode"
msgstr ""

msgid "Category ID scheme"
msgstr ""

msgid "Children"
msgstr ""

msgid "Object type"
msgstr ""

msgid "Compression"
msgstr ""

msgid "Data element ID scheme"
msgstr ""

msgid "Dry run"
msgstr ""

msgid "End date"
msgstr ""

msgid "Event ID Scheme"
msgstr ""

msgid "Flush Mode"
msgstr ""

msgid "Format"
msgstr ""

msgid "ID scheme"
msgstr ""

msgid "Identifier"
msgstr ""

msgid "Report Mode"
msgstr ""

msgid "Import Strategy"
msgstr ""

msgid "Include deleted"
msgstr ""

msgid "Inclusion"
msgstr ""

msgid "Inclusion Strategy"
msgstr ""

msgid "Merge Mode"
msgstr ""

msgid "Object"
msgstr "Objekt"

msgid "Organisation unit"
msgstr "Organizační jednotka"

msgid "Org unit ID scheme"
msgstr ""

msgid "Preheat cache"
msgstr ""

msgid "Preheat Mode"
msgstr ""

msgid "Program Stages"
msgstr ""

msgid "Programs"
msgstr ""

msgid "Schemas"
msgstr ""

msgid "Data Sets"
msgstr ""

msgid "Sharing"
msgstr ""

msgid "Existing record check"
msgstr ""

msgid "Skip Sharing"
msgstr ""

msgid "Skip Validation"
msgstr ""

msgid "Start date"
msgstr ""

msgid "Strategy"
msgstr ""

msgid "Yes"
msgstr "Ano"

msgid "No"
msgstr "Ne"

msgid "UID"
msgstr "UID"

msgid "Code"
msgstr "Kód"

msgid "Name"
msgstr "Název"

msgid "Include descendants of organisation unit"
msgstr ""

msgid "Selected organisation unit"
msgstr ""

msgid "Yes (faster for large imports)"
msgstr ""

msgid "New and Updates"
msgstr ""

msgid "New only"
msgstr ""

msgid "Updates only"
msgstr ""

msgid "Delete"
msgstr "Smazat"

msgid "Skip check (fast)"
msgstr ""

msgid "Check (safe, recommended)"
msgstr ""

msgid "AUTO"
msgstr ""

msgid "Full"
msgstr ""

msgid "Debug"
msgstr ""

msgid "Reference"
msgstr ""

msgid "All"
msgstr "Vše"

msgid "None"
msgstr "Žádný"

msgid "Create and Update"
msgstr ""

msgid "Create"
msgstr ""

msgid "Update"
msgstr "Aktualizovat"

msgid "Merge"
msgstr ""

msgid "Replace"
msgstr ""

msgid "Auto"
msgstr ""

msgid "With sharing"
msgstr ""

msgid "Without sharing"
msgstr ""

msgid "Non Null"
msgstr ""

msgid "Always"
msgstr ""

msgid "Non Empty"
msgstr ""

msgid "Include children of organisation unit"
msgstr ""

msgid "Zip"
msgstr ""

msgid "Gzip"
msgstr ""

msgid "Uncompressed"
msgstr ""

msgid "Category combination"
msgstr "Kombinace kategorie"

msgid "Dashboard"
msgstr "Ovládací panel"

msgid "Data element groups"
msgstr ""

msgid "Option sets"
msgstr ""

msgid "[ All program stages]"
msgstr ""

msgid "Import/Export"
msgstr ""

msgid "Data Export"
msgstr ""

msgid ""
"Export data values. This is the regular export function which exports data "
"to the DHIS 2 exchange format called DXF 2."
msgstr ""

msgid "Loading options..."
msgstr ""

msgid "Event Export"
msgstr ""

msgid ""
"Export event data for programs, stages and tracked entities in the DXF 2 "
"format."
msgstr ""

msgid "Metadata Export"
msgstr ""

msgid ""
"Export meta data like data elements and organisation units to the standard "
"DHIS 2 exchange format."
msgstr ""

msgid "Meta Data Export"
msgstr ""

msgid "Metadata Dependency Export"
msgstr ""

msgid ""
"Export metadata like data sets and programs including related metadata "
"objects."
msgstr ""

msgid "Metadata Export with Dependencies"
msgstr ""

msgid "Import / Export"
msgstr ""

msgid "Import or Export DHIS2 data."
msgstr ""

msgid "Data Import"
msgstr ""

msgid ""
"Import data values on the DXF 2 XML, JSON, CSV and PDF formatrant s. DXF 2 "
"is the standard exchange format for DHIS 2."
msgstr ""

msgid "Event Import"
msgstr ""

msgid ""
"Import events for programs, stages and tracked entities in the DXF 2 format."
msgstr ""

msgid "GML Import"
msgstr ""

msgid ""
"Import geographic data for organisation units using GML format. GML is an "
"XML grammar for expressing geographical features."
msgstr ""

msgid "Only import of GML data for existing organisation units is supported."
msgstr ""

msgid "Metadata Import"
msgstr ""

msgid ""
"Import metadata like data elements and organisation units using the standard"
" DHIS 2 exchange format called DXF 2."
msgstr ""

msgid "It seems like there was no data in the uploaded file."
msgstr ""
