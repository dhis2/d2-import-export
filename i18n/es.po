# 
# Translators:
# phil_dhis2, 2020
# Viktor Varland <viktor@dhis2.org>, 2020
# Marta Vila <marta@dhis2.org>, 2020
# Gabriela Rodriguez <grodriguezcoronado@gmail.com>, 2020
# 
msgid ""
msgstr ""
"Project-Id-Version: i18next-conv\n"
"POT-Creation-Date: 2020-03-16T08:18:48.259Z\n"
"PO-Revision-Date: 2019-06-04 17:24+0000\n"
"Last-Translator: Gabriela Rodriguez <grodriguezcoronado@gmail.com>, 2020\n"
"Language-Team: Spanish (https://www.transifex.com/hisp-uio/teams/100509/es/)\n"
"MIME-Version: 1.0\n"
"Content-Type: text/plain; charset=UTF-8\n"
"Content-Transfer-Encoding: 8bit\n"
"Language: es\n"
"Plural-Forms: nplurals=2; plural=(n != 1);\n"

msgid "Something went wrong when loading the current user!"
msgstr ""

msgid "Something went wrong when loading the data sets!"
msgstr ""

msgid "Filter data sets by name"
msgstr ""

msgid "At least one data set must be selected"
msgstr ""

msgid "One data set must be selected"
msgstr ""

msgid "Invalid date"
msgstr ""

msgid "Start date must be before end date"
msgstr ""

msgid "End date must be after start date"
msgstr ""

msgid ""
"Something went wrong when loading the additional data element ID schemes"
msgstr ""

msgid "Something went wrong when loading the additional ID schemes"
msgstr ""

msgid ""
"Something went wrong when loading the additional organisation unit ID "
"schemes"
msgstr ""

msgid "Please select a file"
msgstr ""

msgid "Will do a test run without importing any data into the database"
msgstr ""

msgid "Dry run"
msgstr ""

msgid "Import"
msgstr "Importar"

msgid "Atomic mode"
msgstr ""

msgid "Object"
msgstr ""

msgid "Compression"
msgstr ""

msgid "Data element ID scheme"
msgstr ""

msgid "Data sets"
msgstr "Sets de datos"

msgid "End date"
msgstr "Fecha final"

msgid "Event ID scheme"
msgstr ""

msgid "Export"
msgstr "Exportar"

msgid "File"
msgstr "Fichero"

msgid "First row is header"
msgstr ""

msgid "Flush mode"
msgstr ""

msgid "Format"
msgstr ""

msgid "ID scheme"
msgstr ""

msgid "Identifier"
msgstr "Identificador"

msgid "Import report mode"
msgstr ""

msgid "Import strategy"
msgstr ""

msgid "Include children"
msgstr ""

msgid "Include deleted"
msgstr ""

msgid "Inclusion"
msgstr ""

msgid "Inclusion strategy"
msgstr ""

msgid "Async"
msgstr ""

msgid "Merge mode"
msgstr ""

msgid "Object type"
msgstr "Tipo de objeto"

msgid "Organisation unit ID scheme"
msgstr ""

msgid "Organisation unit"
msgstr "Unidad organizativa"

msgid "Preheat cache"
msgstr ""

msgid "Faster for large imports"
msgstr ""

msgid "Preheat mode"
msgstr ""

msgid "Programs"
msgstr "Programas"

msgid "Program stage"
msgstr "Etapa de programa"

msgid "Skip audit"
msgstr ""

msgid "Improves performance at the cost of ability to audit changes"
msgstr ""

msgid "Skip exisiting check"
msgstr ""

msgid "Faster, but unsafe"
msgstr ""

msgid "Skip sharing"
msgstr ""

msgid "Skip validation"
msgstr ""

msgid "Start date"
msgstr "Fecha inicial"

msgid "Strategy"
msgstr ""

msgid "No jobs started yet."
msgstr ""

msgid "Recreate job"
msgstr ""

msgid "Completed"
msgstr "Completado"

msgid "In progress"
msgstr "En proceso"

msgid "Job details"
msgstr ""

msgid "Key"
msgstr ""

msgid "Value"
msgstr "Valor"

msgid "Events"
msgstr "Eventos"

msgid "Time"
msgstr ""

msgid "Message"
msgstr "Mensaje"

msgid "ID"
msgstr "Identificador"

msgid "Job summary"
msgstr ""

msgid "Error"
msgstr "Error"

msgid "Conflicts"
msgstr ""

msgid "Messages"
msgstr "Mensajes"

msgid "UID"
msgstr ""

msgid "Type"
msgstr "Tipo"

msgid "Property"
msgstr ""

msgid "View summary of recently started import job"
msgstr ""

msgid "Status"
msgstr "Estado"

msgid "Description"
msgstr "Descripción"

msgid "Summary"
msgstr ""

msgid "Created"
msgstr "Creado"

msgid "Deleted"
msgstr ""

msgid "Ignored"
msgstr ""

msgid "Updated"
msgstr ""

msgid "Total"
msgstr "Total"

msgid "Overview"
msgstr "General"

msgid "Type count"
msgstr ""

msgid "more options"
msgstr ""

msgid "One object must be selected"
msgstr ""

msgid "Something went wrong when loading the organisation units!"
msgstr ""

msgid "At least one organisation unit must be selected"
msgstr "Debe seleccionar al menos una unidad organizativa."

msgid "One organisation unit must be selected"
msgstr ""

msgid "Something went wrong when loading the programs!"
msgstr ""

msgid "Filter programs by name"
msgstr ""

msgid "At least one program must be selected"
msgstr ""

msgid "One program must be selected"
msgstr ""

msgid "One program stage must be selected"
msgstr ""

msgid "Something went wrong when loading the schemas!"
msgstr ""

msgid "Select All"
msgstr ""

msgid "Select None"
msgstr ""

msgid "At least one schema must be selected"
msgstr ""

msgid "Select all"
msgstr "Seleccionar todos"

msgid "Clear all"
msgstr "Clear all"

msgid "Data import"
msgstr ""

msgid "Event import"
msgstr ""

msgid "GML import"
msgstr ""

msgid "Metadata import"
msgstr ""

msgid "Data export"
msgstr ""

msgid "Event export"
msgstr ""

msgid "Metadata dependency export"
msgstr ""

msgid "Metadata export"
msgstr ""

msgid "Job overview"
msgstr ""

msgid "Yes"
msgstr "Si"

msgid "No"
msgstr "No"

msgid "Data"
msgstr "Datos"

msgid "Event"
msgstr "Evento"

msgid "GML"
msgstr ""

msgid "Metadata"
msgstr "Metadatos"

msgid "Something went wrong when loading the class keys"
msgstr ""

msgid "Something went wrong when loading the objects"
msgstr ""

msgid "All program stages"
msgstr ""

msgid "Something went wrong when loading the program stages"
msgstr ""

msgid "Export data values as ADX XML, DFX 2 XML, JSON or CSV files."
msgstr ""

msgid "HTTP error when fetching data"
msgstr ""

msgid "Import data values from ADX XML, DXF 2 XML, JSON, CSV or PDF files."
msgstr ""

msgid ""
"Export event data for programs, stages and tracked entities in the DXF 2 "
"format."
msgstr ""

msgid ""
"Import events for programs, stages and tracked entities using the DXF 2 "
"format."
msgstr ""

msgid ""
"Import geographic data for organisation units using the GML format. GML is "
"an XML grammar for expressing geographical features."
msgstr ""

msgid "An overview of all import jobs started this session."
msgstr ""

msgid ""
"Export metadata like data sets and programs including related metadata "
"objects in the XML or JSON format."
msgstr ""

msgid ""
"Export meta data like data elements and organisation units in the XML, JSON "
"or CSV format."
msgstr ""

msgid ""
"Import metadata like data elements and organisation units using the DXF 2 "
"format."
msgstr ""

msgid "A class key must be selected"
msgstr ""

msgid "An unknown error occurred. Please try again later"
msgstr ""

msgid "JSON"
msgstr "JSON"

msgid "CSV"
msgstr "CSV"

msgid "XML"
msgstr "XML"

msgid "ADX"
msgstr ""

msgid "PDF"
msgstr ""

msgid "Zip"
msgstr ""

msgid "GZip"
msgstr ""

msgid "Uncompressed"
msgstr ""

msgid "Category combination"
msgstr "Combinacion de categoria"

msgid "Dashboard"
msgstr "Tablero"

msgid "Data element groups"
msgstr "Grupos de elementos de datos"

msgid "Option sets"
msgstr ""

msgid "Uid"
msgstr ""

msgid "Code"
msgstr "Código"

msgid "Name"
msgstr "Nombre"

msgid "Selected organisation unit"
msgstr ""

msgid "Include children of organisation unit"
msgstr ""

msgid "Include descendants of organisation unit"
msgstr ""

msgid "Skip check"
msgstr ""

msgid "Check"
msgstr "Ver"

msgid "New and updates"
msgstr ""

msgid "New only"
msgstr ""

msgid "Updates only"
msgstr ""

msgid "Delete"
msgstr "Borrar"

msgid "Auto"
msgstr ""

msgid "Errors"
msgstr "Errores"

msgid "Full"
msgstr ""

msgid "Debug"
msgstr ""

msgid "Reference"
msgstr "Referencia"

msgid "All"
msgstr "Todos"

msgid "None"
msgstr "Ninguno"

msgid "Merge"
msgstr "Mezclar"

msgid "Replace"
msgstr "Reemplazar"

msgid "Non Null"
msgstr ""

msgid "Always"
msgstr ""

msgid "Non Empty"
msgstr ""
