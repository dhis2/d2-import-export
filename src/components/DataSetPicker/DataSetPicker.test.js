import React from 'react'
import i18n from '@dhis2/d2-i18n'
import { render, waitFor, fireEvent } from 'test-utils'
import '@testing-library/jest-dom/extend-expect'
import { CustomDataProvider } from '@dhis2/app-runtime'

import { DataSetPicker } from '../'

const customData = {
    dataSets: {
        dataSets: [
            { id: 'lyLU2wR22tC', displayName: 'ART monthly summary' },
            { id: 'BfMAe6Itzgt', displayName: 'Child Health' },
            { id: 'Lpw6GcnTrmS', displayName: 'Emergency Response' },
        ],
    },
}

const props = {
    selected: [],
    setSelected: () => 1,
    multiSelect: false,
    withFilter: false,
    withActions: false,
    meta: {},
    label: 'Data set',
    dataTest: 'data-set-picker',
}

test('data sets load and are shown in a list', async () => {
    const setSelected = jest.fn()

    const { getByDataTest, getAllByDataTest, getByText } = render(
        <CustomDataProvider data={customData}>
            <DataSetPicker {...props} setSelected={setSelected} />
        </CustomDataProvider>
    )

    const loading = getByDataTest('data-set-picker-loading')
    expect(loading).toBeInTheDocument()

    // loading has finished
    await waitFor(() => getByDataTest('data-set-picker-list'))
    expect(loading).not.toBeInTheDocument()

    const listElements = getAllByDataTest(/data-set-picker-list-body-li-*/)
    expect(listElements).toHaveLength(customData.dataSets.dataSets.length)

    // all data sets are in the document
    customData.dataSets.dataSets.forEach(d => {
        const el = getByDataTest(`data-set-picker-list-body-li-${d.id}`)
        expect(el).toHaveTextContent(d.displayName)
    })

    // clicking on first item fires setSelected
    const firstItem = customData.dataSets.dataSets[0]
    fireEvent.click(getByText(firstItem.displayName))
    expect(setSelected).toHaveBeenCalledTimes(1)
    expect(setSelected).toHaveBeenLastCalledWith([firstItem.id])
})

test('data sets fail to load and an error is shown', async () => {
    // suppress console errors
    console.error = jest.fn()

    const { getByDataTest, getByText } = render(
        <CustomDataProvider data={{}}>
            <DataSetPicker {...props} />
        </CustomDataProvider>
    )

    const loading = getByDataTest('data-set-picker-loading')
    expect(loading).toBeInTheDocument()

    // loading has finished
    await waitFor(() =>
        getByText(i18n.t('Something went wrong when loading the data sets!'))
    )
    expect(loading).not.toBeInTheDocument()
})
