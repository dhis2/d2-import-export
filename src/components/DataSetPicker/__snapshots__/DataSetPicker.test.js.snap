// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-test="data-set-picker"
  >
    <span>
      Data sets
    </span>
    <div>
      <div
        class="jsx-1261481057 "
        data-test="dhis2-uicore-circularloader"
        role="progressbar"
      >
        <svg
          class="jsx-1261481057"
          viewBox="22 22 44 44"
        >
          <circle
            class="jsx-1261481057 circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders filterable list 1`] = `
<DocumentFragment>
  <div
    data-test="data-set-picker"
  >
    <span>
      Data sets
    </span>
    <div>
      <div
        class="jsx-1261481057 "
        data-test="dhis2-uicore-circularloader"
        role="progressbar"
      >
        <svg
          class="jsx-1261481057"
          viewBox="22 22 44 44"
        >
          <circle
            class="jsx-1261481057 circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders multi-select list 1`] = `
<DocumentFragment>
  <div
    data-test="data-set-picker"
  >
    <span>
      Data sets
    </span>
    <div>
      <div
        class="jsx-1261481057 "
        data-test="dhis2-uicore-circularloader"
        role="progressbar"
      >
        <svg
          class="jsx-1261481057"
          viewBox="22 22 44 44"
        >
          <circle
            class="jsx-1261481057 circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders select/clear all actions 1`] = `
<DocumentFragment>
  <div
    data-test="data-set-picker"
  >
    <span>
      Data sets
    </span>
    <div>
      <div
        class="jsx-1261481057 "
        data-test="dhis2-uicore-circularloader"
        role="progressbar"
      >
        <svg
          class="jsx-1261481057"
          viewBox="22 22 44 44"
        >
          <circle
            class="jsx-1261481057 circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
